### PR TITLE
Listing of Wine Recommendations

### PR DESCRIPTION
**Pull request by AnnekeSchuette**
I created a listing of **wine recommandations** as styled components (storybook), including unit tests.
**Thanks in advance for your feedback and improvement suggestions!** 💣🚀🙏🏻

## Value statement
As a **user** 
I **need a listing of wine recommandations**
so that I can  **decide** which wine I want to try

## Acceptance criteria
The listing
- shows multiple cards with different values
- is scrollable

Each WineCard shows the following information:
- Image
- Name of wine
- Description (trimmed, 4 lines max.)
- Price 
- Rating (as number)

### CODE REVIEW

File: client/src/components/WineCard/WineCard.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  font-size: 0.8em;
+  padding: 0 var(--space-small) var(--space-small);
+  display: grid;
+  gap: var(--space-xsmall);
+
+  h2 {
+    font-weight: 400;
+  }
+`
+const InfoWrapper = styled.div``
```
Comment (macfrei) [inline client/src/components/WineCard/WineCard.js:72]: If you don't define any styles, you could just leave it as a `div` or get rid of it completely.

File: client/src/components/WineListing/WineListing.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import WineCard from 'components/WineCard/WineCard'
+import styled from 'styled-components'
```
Comment (macfrei) [inline client/src/components/WineListing/WineListing.js:2]:
```suggestion
import styled from 'styled-components/macro'
```

File: client/src/components/WineListing/WineListing.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import { v4 as uuidv4 } from 'uuid'
+
+export default function WineListing({ results }) {
```
Comment (macfrei) [inline client/src/components/WineListing/WineListing.js:5]: Why is the props called `results`? Are you searching for them first? The name is a bit confusing, I think, but that's just my opinion.

File: client/src/components/WineListing/WineListing.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      description,
+      imageUrl,
+      price,
+      averageRating,
+      ratingCount,
+      score,
+    }) => (
+      <WineCard
+        id={id}
+        key={uuidv4()}
```
Comment (macfrei) [inline client/src/components/WineListing/WineListing.js:19]: Why do you create a new `uuid` for every map and don't use the `id` you already have as  a key?

File: routes/categories.js [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+      ancest.unshift({ _id, name, slug })
+      const category = await Category.findByIdAndUpdate(id, {
+        $set: { ancestors: ancest },
+      })
+    }
+  } catch (err) {
+    console.log(err.message)
+  }
+}
+// create sub categories
```
Comment (macfrei) [inline routes/categories.js:64]: If you don't need this code, you should delete the comments before merging to main :)

**Review by macfrei — COMMENTED** [review]
Good job! I added a few comments though :) 
I didn't expect to see your backend implementation in this PR though. Maybe consider creating a PR just for the backend implementation as it doesn't really belong in this PR. I have looked over it briefly and it looks good, but I would recommend a dedicated PR so that others can have a look at it as well 👍 
Other than that: keep up the good work 🚀

File: client/src/components/WineListing/WineListing.js (hunk 3 above)
Comment (AnnekeSchuette) [reply client/src/components/WineListing/WineListing.js:5]: Exactly, it will be results from a search in next step.

**Comment by AnnekeSchuette** [comment]
@macfrei I made some changes, can you approve? Thanks!
https://github.com/AnnekeSchuette/capstone-project/compare/9ad2812b54ce31e5937d51975e213e61d259aea2...a0899f11110962921691306a0146573d6fd5b800

**Comment by macfrei** [comment]
Looks good :)

**Pull request merged**